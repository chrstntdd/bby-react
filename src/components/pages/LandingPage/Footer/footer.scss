@import '$homeDir/styles/rwd-helpers.scss';
@import '$homeDir/styles/colors.scss';

footer {
  width: 100vw;
  background-color: $BESTBUY_BLUE;
  padding: auto 50px;
  ul {
    padding-top: 1%;
    padding-left: 0;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    margin: 0 auto;
    li {
      @media (min-width: $bp-xxsmall) {
        margin-top: between(9px, 12px, $bp-xxsmall, $bp-xlarge);
        margin-bottom: between(9px, 12px, $bp-xxsmall, $bp-xlarge);
      }
      list-style: none;
      padding-left: 0;
      a {
        letter-spacing: 2px;
        text-transform: uppercase;
        font-size: 1rem;
        text-decoration: none;
        color: white;
        transition-property: color;
        transition-duration: 0.5s;
        &:hover {
          transition-property: color;
          color: $BESTBUY_YELLOW;
        }
      }
    }
  }
}
