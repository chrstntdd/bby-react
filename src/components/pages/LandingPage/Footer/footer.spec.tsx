import * as React from 'react';
import { shallow } from 'enzyme';

import Footer from './';

describe('<Footer/>', () => {
  const props = {
    footerData: [
      {
        name: 'Official Best Buy API Status',
        link: 'https://developer.bestbuy.com/api-status'
      },
      {
        name: 'Sign up',
        link: '/sign-up'
      },
      {
        name: 'Sign in',
        link: '/sign-in'
      }
    ]
  };
  it('should render without crashing', () => {
    const wrapper = shallow(<Footer {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should render 4 items in a list', () => {
    const wrapper = shallow(<Footer {...props} />);
    const listItems = wrapper.find('li');
    expect(listItems.length).toBe(4);
  });
});
