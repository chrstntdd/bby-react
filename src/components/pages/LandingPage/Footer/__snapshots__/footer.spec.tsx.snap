// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render without crashing 1`] = `
<footer>
  <ul>
    <li
      key="0"
    >
      <Link
        replace={false}
        to="https://developer.bestbuy.com/api-status"
      >
        Official Best Buy API Status
      </Link>
    </li>
    <li
      key="1"
    >
      <Link
        replace={false}
        to="/sign-up"
      >
        Sign up
      </Link>
    </li>
    <li
      key="2"
    >
      <a
        href="/sign-in"
        target="_blank"
      >
        Sign in
      </a>
    </li>
    <li>
      <a
        href="https://github.com/chrstntdd"
        target="_blank"
      >
        Created by Christian Todd
      </a>
    </li>
  </ul>
</footer>
`;
